bundles/org.openhab.binding.zm/src/main/java/org/openhab/binding/zm/internal/handler/ZmMonitorHandler.java

```diff
@@ -0,0 +1,328 @@
+        this.timeZoneProvider = timeZoneProvider;
+    }
+
+    @Override
+    public void initialize() {
+        ZmMonitorConfig config = getConfigAs(ZmMonitorConfig.class);
+        monitorId = config.monitorId;
+        logger.debug("Monitor {}: Handler initializing", monitorId);
+        imageRefreshInterval = config.imageRefreshInterval;
+        Integer value = config.alarmDuration;
+        alarmDuration = value != null ? value : DEFAULT_ALARM_DURATION_SECONDS;
+        updateMonitorStatus();
+        startImageRefreshJob();
+    }
+
+    @Override
+    public void dispose() {
+        logger.debug("Monitor {}: Handler disposing", monitorId);
+        stopAlarmOffJob();
+        turnAlarmOff();
+        stopImageRefreshJob();
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        if (command instanceof RefreshType) {
+            // No need to handle as status will be updated at the next refresh interval
+            return;
+        }
+        logger.debug("Monitor {}: Received command '{}' for channel '{}'", monitorId, command, channelUID.getId());
+        ZmBridgeHandler localHandler = bridgeHandler;
+        if (localHandler == null) {
+            logger.info("Monitor {}: Can't execute command because bridge handler is null", monitorId);
+            return;
+        }
+        switch (channelUID.getId()) {
+            case CHANNEL_FUNCTION:
+                if (command instanceof StringType) {
+                    try {
+                        MonitorFunction function = MonitorFunction.forValue(command.toString());
+                        localHandler.setFunction(monitorId, function);
+                        logger.debug("Monitor {}: Set monitor state to {}", monitorId, function);
+                    } catch (IllegalArgumentException e) {
+                        logger.debug("Monitor {}: Invalid function: {}", monitorId, command);
+                    }
+                }
+                break;
+            case CHANNEL_ENABLE:
+                if (command instanceof OnOffType) {
+                    localHandler.setEnabled(monitorId, (OnOffType) command);
+                    logger.debug("Monitor {}: Set monitor enable to {}", monitorId, command);
+                }
+                break;
+            case CHANNEL_TRIGGER_ALARM:
+                if (command instanceof OnOffType) {
+                    logger.debug("Monitor {}: Set monitor alarm to {}", monitorId, command);
+                    if (command == OnOffType.ON) {
+                        localHandler.setAlarmOn(monitorId);
+                        startAlarmOffJob(alarmDuration.intValue());
+                    } else {
+                        stopAlarmOffJob();
+                        localHandler.setAlarmOff(monitorId);
+                    }
+                }
+                break;
+        }
+    }
+
+    @Override
+    public void bridgeStatusChanged(ThingStatusInfo bridgeStatusInfo) {
```

Comment:
Is this method necessary? Child things are updated by the framework if the bridge becomes offline.

Reply:
IDK. I thought this was necessary in order for the child handlers to be informed when the bridge status changes. It is used pretty widely across many bindings.

Reply:
Maybe I miss something. @cpmeister Is there any reason why the bridge should update its child states to offline when going offline itself?

Reply:
I tested it: Things are automatically set offline, when the bridge goes offline. So, this method could be removed.

Reply:
Changed